src/App.tsx

```diff
@@ -1,17 +1,113 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './app.typedefs';
+import todos from './api/todos';
 import users from './api/users';
 
+const preparedTodos: PreparedTodo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || undefined,
+}));
+
 const App: React.FC = () => {
+  const [todosList, setTodos] = useState(preparedTodos);
+  const [todosTitle, setTodosTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const addTodo = (title: string, userId: number) => {
+    const todo: PreparedTodo = {
+      userId,
+      id: todosList.length + 1,
+      title,
+      completed: false,
+      user: users.find(user => user.id === userId),
+    };
+
+    setTodos([...todosList, todo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent):void => {
+    event.preventDefault();
+
+    if (!todosTitle && !selectedUser) {
+      setTitleError(true);
+      setUserError(true);
+    } else if (!selectedUser) {
+      setUserError(true);
+    } else if (!todosTitle) {
+      setTitleError(true);
```

Comment:
this whole `if ... else if ... else` chain is not graceful
think of if from this perspective - how do you return a boolean whether variable is not falsy? (`!undefined = true` ; `!'string' = false`) so you could just `setTitleError(!title)` - same for the user
and you won't need the entire `if ... else if ... else` chain